querybook/server/datasources/board.py

```diff
@@ -186,11 +200,58 @@ def move_board_item(board_id, from_index, to_index):
 )
 def delete_board_item(board_id, item_type, item_id):
     api_assert(
-        item_type == "data_doc" or item_type == "table" or item_type == "board",
+        item_type == "data_doc"
+        or item_type == "table"
+        or item_type == "board"
+        or item_type == "query",
         "Invalid item type",
     )
     with DBSession() as session:
         assert_can_edit(board_id, session=session)
 
         board = Board.get(id=board_id, session=session)
         logic.remove_item_from_board(board.id, item_id, item_type, session=session)
+
+
+@register("/board/favorite/", methods=["POST"])
+def get_or_create_favorite_board(environment_id):
+    verify_environment_permission([environment_id])
+    with DBSession() as session:
+        board = logic.get_or_create_user_favorite_board(
+            current_user.id, environment_id, session=session
+        )
+        return board.to_dict(
+            extra_fields=["docs", "tables", "boards", "queries", "items"]
+        )
+
+
+@register(
+    "/board/<int:board_id>/item/<int:board_item_id>/description/", methods=["PUT"]
+)
+def update_board_item_description(board_id, board_item_id, description):
+    with DBSession() as session:
+        assert_can_edit(board_id, session=session)
```

Comment:
board id doesn't need to be associated with board_item_id, some hacker could pass in a board id they own and board_item_id that they don't.

We can infer board_id from board_item_id instead